website/src/components/Visualisations/IIGOPayments/Util/ProcessedPaymentsData.ts

```diff
@@ -0,0 +1,116 @@
+import { OutputJSONType, TeamName } from '../../../../consts/types'
+import { ProcessedTaxData, TaxData } from './IIGOPaymentsTypes'
+
+export const processPaymentsData = (data: OutputJSONType): ProcessedTaxData => {
+    if (data.GameStates.length === 0) return []
+
+    const retData: ProcessedTaxData = []
+    const IIGOHistory = Object.entries(
+        data.GameStates[data.GameStates.length - 1].IIGOHistory
+    )
+
+    const empty = (): TaxData => {
+        return {
+            expectedAlloc: 0,
+            actualAlloc: 0,
+            expectedTax: 0,
+            actualTax: 0,
+            expectedSanction: 0,
+            actualSanction: 0,
+        }
+    }
+
+    const TaxInfo: Map<TeamName, TaxData> = new Map<TeamName, TaxData>([
+        [1, empty()],
+        [2, empty()],
+        [3, empty()],
+        [4, empty()],
+        [5, empty()],
+        [6, empty()],
+    ])
```

Comment:
does anyone else know a cleaner way of doing this? :O 

I would do it in a different way in which I `map` then `reduce`, but hey, if this works, it works :D

Reply:
I would have done it with map and reduce too can't think of a nicer way :O if it works it works :D

Reply:
so should i change it to map and reduce, or should i just keep it this way since its (kind of) working?

Reply:
keep it its fine we have basically no test coverage on the front end anyway it's fine for this sort of scale project I think :)